coffea/processor/executor.py

```diff
@@ -817,28 +1029,143 @@ def __call__(
             parsl.clear()
             parsl.load(self.config)
 
-        app = timeout(python_app(function))
+        # Check config/executors
+        _exec_avail = [exe.label for exe in parsl.dfk().config.executors]
+        _execs_tried = (
+            [] if self.jobs_executors == "all" else [e for e in self.jobs_executors]
+        )
+        _execs_tried += (
+            [] if self.merges_executors == "all" else [e for e in self.merges_executors]
+        )
+        if not all([_e in _exec_avail for _e in _execs_tried]):
+            raise RuntimeError(
+                f"Executors: [{','.join(_e for _e in _execs_tried if _e not in _exec_avail)}] not available in the config."
+            )
+
+        # Apps
+        app = timeout(python_app(function, executors=self.jobs_executors))
+        reducer = timeout(
+            python_app(_reduce(self.compression), executors=self.merges_executors)
+        )
+
+        # Initiate
+        FH = FuturesHolder(set(map(app, items)), refresh=2)
+        nparts, minred, maxred = 5, 4, 100
+        if isinstance(self.merging, tuple):
+            if len(self.merging) == 3:
+                nparts, minred, maxred = self.merging
 
-        gen = _futures_handler(map(app, items), self.tailtimeout)
         try:
-            accumulator = accumulate(
-                tqdm(
-                    gen if self.compression is None else map(_decompress, gen),
-                    disable=not self.status,
-                    unit=self.unit,
-                    total=len(items),
-                    desc=self.desc,
-                ),
-                accumulator,
-            )
-        finally:
-            gen.close()
+            with rich_bar() as progress:
```

Comment:
How much of this section is different from the `FuturesExecutor` impl? It looks quite similar, can we deduplicate? (maybe by putting the future update loop into FuturesHolder and passing it a callback to submit the processing and reducing task?)

Reply:
refactored